### PR TITLE
Fixed user being head and headDeputy at the same

### PR DESCRIPTION
**Pull request by andrii52**
time

dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by OleksandrBilan — APPROVED** [review]

File: src/pages/City/AddAdministratorModal/AddCitiesSecretaryForm.tsx [hunk 1]
```diff
@@ -47,22 +51,35 @@ const AddCitiesNewSecretaryForm = (props: any) => {
...
+      user: value.user,
+      endDate: value.endDate,
+      startDate: value.startDate,
     };
-    onAdd(newAdmin);
+    return admin;
+  }
+
+  const handleSubmit = async (values: any) => {
+    if(JSON.parse(values.userId).id == props.head?.userId ) {
```
Comment (OleksandrBilan) [inline src/pages/City/AddAdministratorModal/AddCitiesSecretaryForm.tsx:73]: put space after if

File: src/pages/City/AddAdministratorModal/AddCitiesSecretaryForm.tsx [hunk 2]
```diff
@@ -47,22 +51,35 @@ const AddCitiesNewSecretaryForm = (props: any) => {
...
+      const newAdmin = SetAdmin(props.head, values);
+      onAdd(newAdmin);
+    }else if(JSON.parse(values.userId).id == props.headDeputy?.userId){
```
Comment (OleksandrBilan) [inline src/pages/City/AddAdministratorModal/AddCitiesSecretaryForm.tsx:76]: put space before else

File: src/pages/City/City/City.tsx [hunk 3]
```diff
@@ -302,28 +302,39 @@ const City = () => {
...
+    }
+  }
+
+  const handleOk = async(admin: CityAdmin) => {
+    try {
+      if (admin.adminType.adminTypeName === Roles.CityHead) {
+        if (city.head == null) {
+          
+          checkAdminId(admin);
+        }else {
```
Comment (OleksandrBilan) [inline src/pages/City/City/City.tsx:319]: put space before else

File: src/pages/City/City/City.tsx [hunk 4]
```diff
@@ -302,28 +302,39 @@ const City = () => {
...
-            editCityAdmin(admin);
-          }
-        } else {
-          if (admin.id === 0) {
-            addCityAdmin(admin);
-          }
-          else {
-            editCityAdmin(admin);
+            showConfirmCityAdmin(admin);
+          }else {
```
Comment (OleksandrBilan) [inline src/pages/City/City/City.tsx:322]: and space here

File: src/pages/City/City/City.tsx [hunk 5]
```diff
@@ -302,28 +302,39 @@ const City = () => {
...
+            checkAdminId(admin);
           }
         }
-      } finally {
-        setvisible(false);
+      } else if(admin.adminType.adminTypeName === Roles.CityHeadDeputy) {
+        if (city.headDeputy == null) {
+          checkAdminId(admin);
+        }else{
```
Comment (OleksandrBilan) [inline src/pages/City/City/City.tsx:329]: and here

File: src/pages/City/City/City.tsx [hunk 6]
```diff
@@ -302,28 +302,39 @@ const City = () => {
...
+          checkAdminId(admin);
+        }
+      }
+      else {
```
Comment (OleksandrBilan) [inline src/pages/City/City/City.tsx:333]: else should be in the same row as the '}'

File: src/pages/Club/AddAdministratorModal/AddClubsSecretaryForm.tsx [hunk 7]
```diff
@@ -61,99 +49,43 @@ const AddClubsNewSecretaryForm = (props: any) => {
...
-        if (admin.id === 0) {
-          addClubAdmin(admin);
-        } else {
-          editClubAdmin(admin);
-        }
-      }
-    } finally {
-      onAdd();
+  const handleSubmit = async (values: any) => {
+    if(JSON.parse(values.userId).id == props.head?.userId ) {
```
Comment (OleksandrBilan) [inline src/pages/Club/AddAdministratorModal/AddClubsSecretaryForm.tsx:71]: space

File: src/pages/Club/AddAdministratorModal/AddClubsSecretaryForm.tsx [hunk 8]
```diff
@@ -61,99 +49,43 @@ const AddClubsNewSecretaryForm = (props: any) => {
...
+      const newAdmin = SetAdmin(props.head, values);
+      onAdd(newAdmin);
+    }else if(JSON.parse(values.userId).id == props.headDeputy?.userId){
```
Comment (OleksandrBilan) [inline src/pages/Club/AddAdministratorModal/AddClubsSecretaryForm.tsx:74]: space

File: src/pages/Club/Club/Club.tsx [hunk 9]
```diff
@@ -196,14 +202,110 @@ const Club = () => {
...
+    }
+  }
+
+  const handleOk = async(admin: ClubAdmin) => {
+    console.log(admin.adminType, admin.id)
+    try {
+      if (admin.adminType.adminTypeName === Roles.KurinHead) {
+        if (club.head == null) {
+          checkAdminId(admin);
+        }else {
```
Comment (OleksandrBilan) [inline src/pages/Club/Club/Club.tsx:286]: space

File: src/pages/Club/Club/Club.tsx [hunk 10]
```diff
@@ -196,14 +202,110 @@ const Club = () => {
...
+          if (club.head?.userId !== admin.userId) {
+            showConfirmClubAdmin(admin);
+          }else {
```
Comment (OleksandrBilan) [inline src/pages/Club/Club/Club.tsx:289]: space

File: src/pages/Club/Club/Club.tsx [hunk 11]
```diff
@@ -196,14 +202,110 @@ const Club = () => {
...
+            checkAdminId(admin);
+          }
+        }
+      } else if(admin.adminType.adminTypeName === Roles.KurinHeadDeputy) {
```
Comment (OleksandrBilan) [inline src/pages/Club/Club/Club.tsx:293]: space after if

File: src/pages/Club/Club/Club.tsx [hunk 12]
```diff
@@ -196,14 +202,110 @@ const Club = () => {
...
+        if (club.headDeputy == null) {
+          checkAdminId(admin);
+        }else{
```
Comment (OleksandrBilan) [inline src/pages/Club/Club/Club.tsx:296]: space

File: src/pages/Club/Club/Club.tsx [hunk 13]
```diff
@@ -196,14 +202,110 @@ const Club = () => {
...
+          checkAdminId(admin);
+        }
+      }
+      else {
```
Comment (OleksandrBilan) [inline src/pages/Club/Club/Club.tsx:300]: else should be on row 299

File: src/pages/Regions/AddRegionSecretaryForm.tsx [hunk 14]
```diff
@@ -47,59 +50,55 @@ const AddNewSecretaryForm = (props: any) => {
...
-        `/regions/${currentRegion}`,
-        `цій окрузі`
-      );
-      onAdd();
+    return admin;
+  }
+
+  const handleSubmit = async (values: any) => {
+    console.log(values);
+    if(JSON.parse(values.userId).id == props.head?.userId ) {
```
Comment (OleksandrBilan) [inline src/pages/Regions/AddRegionSecretaryForm.tsx:76]: space

File: src/pages/Regions/AddRegionSecretaryForm.tsx [hunk 15]
```diff
@@ -47,59 +50,55 @@ const AddNewSecretaryForm = (props: any) => {
...
+      const newAdmin = await SetAdmin(props.head, values);
+      console.log(newAdmin)
+      onAdd(newAdmin);  
+    }else if(JSON.parse(values.userId).id == props.headDeputy?.userId){
```
Comment (OleksandrBilan) [inline src/pages/Regions/AddRegionSecretaryForm.tsx:80]: spaces before else and after if

File: src/pages/Regions/AddRegionSecretaryForm.tsx [hunk 16]
```diff
@@ -47,59 +50,55 @@ const AddNewSecretaryForm = (props: any) => {
...
+      onAdd(newAdmin);
+    }
+  };
+  const fetchData = async () => {
+    await regionsApi.getAdminTypes().then((response) => {
+      setTypes(response.data);
+    });
+    if(props.regionId !== undefined)
+    {
+    await regionsApi.getRegionUsers(props.regionId).then((response) => {
```
Comment (OleksandrBilan) [inline src/pages/Regions/AddRegionSecretaryForm.tsx:96]: tabulate these 2 rows

File: src/pages/Regions/Region.tsx [hunk 17]
```diff
@@ -253,15 +225,118 @@ const Region = () => {
...
+  const handleConfirm = async () => {
+    setVisible(false);
+  };
+
+  const handleOk = async(admin: any) => {
+    try {
+      if (admin.adminType.adminTypeName === Roles.OkrugaHead) {
+        if (head == ' ') {
+          checkAdminId(admin);
+        }else {
```
Comment (OleksandrBilan) [inline src/pages/Regions/Region.tsx:316]: space

File: src/pages/Regions/Region.tsx [hunk 18]
```diff
@@ -253,15 +225,118 @@ const Region = () => {
...
+          if (head.userId !== admin.userId) {
+            showConfirmClubAdmin(admin);
+          }else {
```
Comment (OleksandrBilan) [inline src/pages/Regions/Region.tsx:319]: space

File: src/pages/Regions/Region.tsx [hunk 19]
```diff
@@ -253,15 +225,118 @@ const Region = () => {
...
+            checkAdminId(admin);
+          }
+        }
+      } else if(admin.adminType.adminTypeName === Roles.OkrugaHeadDeputy) {
```
Comment (OleksandrBilan) [inline src/pages/Regions/Region.tsx:323]: space after if

File: src/pages/Regions/Region.tsx [hunk 20]
```diff
@@ -253,15 +225,118 @@ const Region = () => {
...
+        if (headDeputy == 'null') {
+          checkAdminId(admin);
+        }else{
```
Comment (OleksandrBilan) [inline src/pages/Regions/Region.tsx:326]: space

File: src/pages/Regions/Region.tsx [hunk 21]
```diff
@@ -253,15 +225,118 @@ const Region = () => {
...
+          checkAdminId(admin);
+        }
+      }
+      else {
```
Comment (OleksandrBilan) [inline src/pages/Regions/Region.tsx:330]: else on previous row

**Review by OleksandrBilan — CHANGES_REQUESTED** [review]

**Review by OleksandrBilan — APPROVED** [review]

**Review by christykobyletska — APPROVED** [review]

**Review by sharix4 — APPROVED** [review]

**Pull request merged**